src/Components/Web.JS/src/Services/WebRootComponentManager.ts

```diff
@@ -32,48 +33,52 @@ type RootComponentRemoveOperation = {
 };
 
 type RootComponentInfo = {
+  descriptor: ComponentDescriptor;
   assignedRendererId?: WebRendererId;
   uniqueIdAtLastUpdate?: number;
   interactiveComponentId?: number;
 }
 
 export class WebRootComponentManager implements DescriptorHandler, NavigationEnhancementCallbacks, RootComponentManager<never> {
-  private readonly _activeDescriptors = new Set<ComponentDescriptor>();
+  private readonly _rootComponents = new Set<RootComponentInfo>();
 
-  private readonly _descriptorsPendingInteractivityById: { [id: number]: ComponentDescriptor } = {};
+  private readonly _descriptors = new Set<ComponentDescriptor>();
 
-  private readonly _rootComponentInfoByDescriptor = new Map<ComponentDescriptor, RootComponentInfo>();
+  private readonly _pendingComponentsToResolve = new Map<number, RootComponentInfo>();
 
-  private _hasPendingRootComponentUpdate = false;
-
-  private _hasStartedCircuit = false;
-
-  private _hasStartedLoadingWebAssembly = false;
-
-  private _hasLoadedWebAssembly = false;
```

Comment:
These "status" flags were moved to `Boot.Server.Common.ts` and `Boot.WebAssembly.Common.ts` so that there's less of a risk of them getting out of sync with the actual loading/started state of each platform.